quinn-h3/src/qpack/prefix_string/decode.rs

```diff
@@ -78,6 +75,23 @@ impl HuffmanDecoder {
     }
 }
 
+fn extract_bits(src: &[u8], mut byte_offset: u32, mut bit_offset: u32, len: u32) -> Result<u8, ()> {
```

Comment:
As someone who's also not a bit twiddling expert, I think this needs a decent battery of unit tests regardless of my opinion of the implementation.

Reply:
Both `extract_bits()` and `set_bits()` are [fully covered](https://codecov.io/gh/djc/quinn/commit/4f5f85daeac3d8f12825e9ea31ea5a612897584f) by existing tests. As such, I don't see a whole lot of value in adding more. What do you think?

Reply:
That link doesn't work here (and I can't seem to find equivalent information by browsing manually) but I think it's somewhat beside the point. It's useful to have narrowly scoped unit tests even if they're subsumed by larger ones, because it's easier to be sure that the narrowly scoped tests cover all interesting cases (which can be more subtle than just ensuring that every branch is traversed), and if something *is* wrong, the narrow tests are much more helpful for diagnosing an issue, both now and in the future (I'm reluctant to bet that we'll never want to touch a given piece of code again).

Reply:
Done.